app.js

```diff
@@ -49,6 +49,8 @@ io.sockets.on("connection", function (socket) {
     socket.emit("handshake", { version: version });
     socket.on("extractMetadata", function (data) {
         if (!data.url) return socket.emit("exception", { message: "URL not provided." });
+        if (!(/^https?:\/\/www.w3.org\/TR\/.+/.test(data.url)))
+            return socket.emit("exception", { message: "Invalid URL of document." });
```

Comment:
Although the check seems fine, it doesn't handle all the use cases:

1. Someone can still try to check the following document (https://www.w3.org/TR/2019/NOTE-wai-aria-practices-1.1-20190207/examples/listbox/listbox-rearrangeable.html) and the exception will break pubrules as it's not the main page of a specification.
2. People may want to check their document *before* uploading them to /TR. In this case, it would be staged somewhere else (like github pages) and pubrules will not run because of the check you added.

You need to check what exception pubrules is generating when it's checking a non-TR doc, catch it and return the error.